operator/v1/types_ingress.go

```diff
@@ -0,0 +1,119 @@
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ClusterIngress describes a discrete, managed ingress component for the
+// cluster. This includes a router, and may also include a load balancer and
+// external wildcard DNS records for that load balancer.
+type ClusterIngress struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ClusterIngressSpec   `json:"spec,omitempty"`
+	Status ClusterIngressStatus `json:"status,omitempty"`
+}
+
+// ClusterIngressSpec is the specification of the desired behavior of the
+// ClusterIngress.
+type ClusterIngressSpec struct {
+	// IngressDomain is a DNS name suffix serviced by the ClusterIngress. If
+	// specified, this value:
+	//
+	// * Is added automatically to the status of Routes to inform downstream systems.
+	// * May be used for a wildcard DNS alias to any provisioned load balancer.
+	// * May be used to generate a default wildcard certificate.
+	//
+	// If nil, the default value from the cluster ingress config is used.
+	//
+	// Once set, this field cannot be changed.
+	IngressDomain *string `json:"ingressDomain"`
+
+	// Replicas is the desired number of router instances.
+	Replicas int32 `json:"replicas"`
```

Comment:
Should be `*int32` and have the same rules as replica sets.  If unset, the ingress controller should manage that.

Reply:
Fixed